webapp/src/components/modals/create_update_issue/create_update_issue.jsx

```diff
@@ -46,17 +46,32 @@ export default class CreateIssueModal extends PureComponent {
         this.validator = new Validator();
     }
 
+    /* eslint-disable react/no-did-update-set-state*/
     componentDidUpdate(prevProps) {
-        if (this.props.post && !prevProps.post) {
-            this.setState({issueDescription: this.props.post.message}); //eslint-disable-line react/no-did-update-set-state
-        } else if (this.props.channelId && (this.props.channelId !== prevProps.channelId || this.props.title !== prevProps.title)) {
-            const title = this.props.title.substring(0, MAX_TITLE_LENGTH);
-            this.setState({issueTitle: title}); // eslint-disable-line react/no-did-update-set-state
+        const data = this.props.messageData;
+        if (this.props.post && !prevProps.post && !data?.title) {
+            this.setState({issueDescription: this.props.post.message});
+        } else if (data?.channel_id && (data?.channel_id !== prevProps.messageData?.channel_id || data?.title !== prevProps.messageData?.title || data?.description !== prevProps.messageData?.description || data?.assignees !== prevProps.messageData?.assignees || data?.labels !== prevProps.messageData?.labels || data?.milestone_title !== prevProps.messageData?.milestone_title || data?.milestone_number !== prevProps.messageData?.milestone_number)) {
+            if (data?.assignees) {
+                this.setState({assignees: data?.assignees});
+            }
+            if (data?.labels) {
+                this.setState({labels: data?.labels});
+            }
+            this.setState({milestone: {
+                value: data?.milestone_number,
+                label: data?.milestone_title,
+            }});
+            this.setState({issueDescription: data?.description});
+            this.setState({repo: data?.repo_full_name});
+            this.setState({issueTitle: data?.title.substring(0, MAX_TITLE_LENGTH)});
         }
     }
+    /* eslint-enable */
 
     // handle issue creation after form is populated
     handleCreate = async (e) => {
+        const data = this.props.messageData;
```

Comment:
do this everywhere wherever possible
```suggestion
        const {channel_id,issue_number} = this.props.messageData;
```